hibernate-core/src/test/java/org/hibernate/orm/test/batchfetch/EagerManyToOneStreamTest.java

```diff
@@ -0,0 +1,200 @@
+package org.hibernate.orm.test.batchfetch;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.hibernate.Hibernate;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.query.spi.QueryImplementor;
+
+import org.hibernate.testing.jdbc.SQLStatementInspector;
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.JiraKey;
+import org.hibernate.testing.orm.junit.ServiceRegistry;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.orm.junit.Setting;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+@DomainModel(
+		annotatedClasses = { EagerManyToOneStreamTest.Child.class, EagerManyToOneStreamTest.Parent.class }
+)
+@SessionFactory( statementInspectorClass = SQLStatementInspector.class)
+@ServiceRegistry(settings = @Setting(name = AvailableSettings.DEFAULT_BATCH_FETCH_SIZE, value = "2"))
+@JiraKey("HHH-15449")
+public class EagerManyToOneStreamTest {
+
+	public static final String FIELD_VALUE = "a field";
+	public static final String FIELD_VALUE_2 = "a second field";
+
+	@BeforeEach
+	public void setUp(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					Parent parent = new Parent( FIELD_VALUE );
+					session.persist( parent );
+					session.persist( new Child( parent ) );
+					Parent parent2 = new Parent( FIELD_VALUE_2 );
+					session.persist( parent2 );
+					session.persist( new Child( parent2 ) );
+				}
+		);
+	}
+
+	@AfterEach
+	public void tearDown(SessionFactoryScope scope) {
+		scope.inTransaction(
+				session -> {
+					session.createMutationQuery( "delete from Child" ).executeUpdate();
+					session.createMutationQuery( "delete from Parent" ).executeUpdate();
+				}
+		);
+	}
+
+	@Test
+	public void testGetResultStreamCollectSingleResult(SessionFactoryScope scope) {
+		final SQLStatementInspector sqlStatementInterceptor = (SQLStatementInspector) scope.getStatementInspector();
+		sqlStatementInterceptor.clear();
+		scope.inTransaction(
+				session -> {
+					QueryImplementor<Child> query = session
+							.createQuery( "select c from Child as c where c.parent.someField=:someField", Child.class )
+							.setParameter( "someField", FIELD_VALUE );
+					Stream<Child> resultStream = query.getResultStream();
+
+					List<Child> children = resultStream.collect( Collectors.toList() );
```

Comment:
would propose checking the entries as they are streamed, rather than doing a collect first. Would catch issues related to differences in streams being handled as entries are read vs. after everything is read (which would be like getResultList)

Reply:
added `testGetResultStreamForEach`